modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -373,110 +373,130 @@ def waiver_boolean(filed5103_waiver_ind)
-              uploads_allowed: # EVSS generates this field
-            }
+          wwr.map do |item|
+            status = accepted?(claim_status_type) ? 'ACCEPTED' : 'INITIAL_REVIEW_COMPLETE'
+            uploaded = true
+
+            build_tracked_item(find_tracked_item(item[:dvlpmt_item_id]), status, item, uploaded)
+          end
+        end
+
+        def build_no_longer_needed_items
+          no_longer_needed = @tracked_items.select do |item|
+            item[:accept_dt].present? && item[:dvlpmt_tc] == 'CLMNTRQST'
+          end
+          return [] if no_longer_needed.empty?
+
+          no_longer_needed.map do |tracked_item|
+            status = 'NO_LONGER_REQUIRED'
+            uploaded = supporting_document?(tracked_item[:dvlpmt_item_id])
+
+            build_tracked_item(tracked_item, status, {}, uploaded)
           end
         end
 
-        def tracked_item_req_date(ebenefits_details, item, id)
-          items = ([ebenefits_details[:wwsnfy]].flatten |
-          [ebenefits_details[:wwr]].flatten |
-          [ebenefits_details[:wwd]].flatten).compact
+        def uploads_allowed?(status)
+          %w[NEEDED_FROM_YOU NEEDED_FROM_OTHERS SUBMITTED_AWAITING_REVIEW INITIAL_REVIEW_COMPLETE].include? status
+        end
 
-          tracked_item = items.find { |i| i[:dvlpmt_item_id] == id } || {}
-          date_present(tracked_item[:date_open] || item[:req_dt] || item[:create_dt])
+        def accepted?(status)
+          ['Preparation for Decision', 'Pending Decision Approval', 'Preparation for Notification',
+           'Complete'].include? status
         end
 
-        def tracked_item_description(tracked_items, id)
-          return nil if tracked_items.nil?
+        def overdue?(tracked_item)
+          overdue_date_present?(tracked_item) ? tracked_item[:suspns_dt] < Time.zone.now : false
+        end
 
-          tracked_item = tracked_items.find { |a| a[:dvlpmt_item_id] == id }
-          tracked_item.present? ? tracked_item[:items] : nil
+        def overdue_date_present?(tracked_item)
+          tracked_item[:suspns_dt].present? && tracked_item[:accept_dt].nil?
+        end
+
+        def build_tracked_item(tracked_item, status, item, uploaded)
+          uploads_allowed = uploads_allowed?(status)
+          {
+            closed_date: date_present(tracked_item[:accept_dt]),
+            description: item[:items],
+            display_name: tracked_item[:short_nm],
+            overdue: overdue?(tracked_item), # EVSS generates this field
+            received_date: date_present(tracked_item[:receive_dt]),
+            requested_date: tracked_item_req_date(tracked_item, item),
+            status:, # EVSS generates this field
+            suspense_date: date_present(tracked_item[:suspns_dt]),
+            id: tracked_item[:dvlpmt_item_id].to_i,
+            uploaded:, # EVSS generates this field
+            uploads_allowed: # EVSS generates this field
```

Comment:
Same question on this comment, we're determining this via uploads_allowed?(status).